router/src/server.rs

```diff
@@ -682,6 +731,8 @@ async fn generate_stream_with_callback(
                                         tracing::debug!(parent: &span, "Output: {}", output_text);
                                         tracing::info!(parent: &span, "Success");
 
+                                        request_logger_sender.send((generated_text.generated_tokens as i64, api_token.unwrap_or("".to_string()), info.model_id.clone())).await;
```

Comment:
Any way we can make the `request_logger_sender` Optional so we don't do this when request logging is disabled? I'm not sure it will cause any issues, but it would be one less vector for things to go wrong (if, for example, we start accumulating a large queue of messages that never gets consumed, etc.).